crates/cow-amm/src/amm.rs

```diff
@@ -12,6 +12,7 @@ use {
 
 #[derive(Clone, Debug)]
 pub struct Amm {
+    contract: contracts::CowAmm,
```

Comment:
If we introduce the contract as a new field we might as well drop the `address` field.

Reply:
moved outside